frontend/src/routes/(authenticated)/user/+page.svelte

```diff
@@ -3,14 +3,25 @@
   import { Button, Form, FormError, Input, lexSuperForm } from '$lib/forms';
   import t from '$lib/i18n';
   import { Page } from '$lib/layout';
-  import { notifySuccess } from '$lib/notify';
+  import { _changeUserAccountData } from './+page';
+  import { notifySuccess, notifyWarning } from '$lib/notify';
   import z from 'zod';
+  import { goto } from '$app/navigation';
+  import DeleteUserModal from '$lib/components/DeleteUserModal.svelte';
   import type { PageData } from './$types';
-  import { _changeUserAccountData } from './+page';
-
+  import { TrashIcon } from '$lib/icons';
+  import {onMount} from 'svelte';
   export let data: PageData;
-  const user = data.user;
+  $: user = data?.user;
+  $: userid = user?.id;
+  let deleteModal: DeleteUserModal;
 
+
+  async function openDeleteModal(): Promise<void> {
+    await deleteModal.open(userid);
+    notifyWarning($t('account_settings.delete_success'))
```

Comment:
This will currently get called even if the user clicks cancel on the modal. You should be able to use the return value from `open` to see if they cancelled or not.

Reply:
Thanks, it should be sorted out now